cli/tests/containerize.bats

```diff
@@ -415,6 +440,32 @@ EOF
   assert_output --partial "Hello, world!"
 }
 
+@test "container can run as specified user and group" {
+  skip_if_not_linux # config is implemented in the Linux build of flox entirely
+  
+  "$FLOX_BIN" init
+
+  MANIFEST_CONTENTS="$(cat << "EOF"
+    version = 1
+    [install]
+    hello.pkg-path = "hello"
+
+    [containerize.config]
+    user = "hello:hello"
+EOF
+  )"
+
+  echo "$MANIFEST_CONTENTS" | _FLOX_USE_CATALOG_MOCK="$GENERATED_DATA/resolve/hello.json" "$FLOX_BIN" edit -f -
+
+  TAG="whoami-in-container"
+
+  bash -c "FLOX_CONTAINERIZE_FLAKE_REF_OR_REV=main $FLOX_BIN containerize --tag $TAG --runtime podman" 3>&- # TODO: why close FD 3?
+
+  run podman run --rm "test:$TAG" 'whoami'
+  assert_success
+  assert_output --partial "hello"
+}
+
```

Comment:
question blocking: do you think we should add coverage for some of the other cases? `user`, `uid`, `uid:gid`, `uid:group`, and `user:gid`? I'm not sure if an integration test would be too expensive. `nixStoreUserGroup` feels like something that would be nice to unit test but I'm not sure exactly how we would do that. Did you at least test manually?

Reply:
I did test these manually, yes. You just have to be careful when running `docker run --user <user:group>` that the `<user:group>` argument matches what you typed in for the `User` config. We could maybe use [runTests](https://nixos.org/manual/nixpkgs/stable/#auto-generated-14-1.11.3) to check these?

Reply:
Something like this?

```console
$ cat test.nix
{
  lib,
}:
lib.debug.runTests {
  testUserConfigMatch = {
    expr = (
      builtins.match "^(([_]*[[:alpha:]]+):?|([[:digit:]]+):?)(([_]*[[:alpha:]]+)|([[:digit:]]+))?$" "foo:1234"
    );
    expected = [
      "foo:"
      "foo"
      null
      "123"
      null
      "123"
    ];
  };
}
$ nix eval --impure --expr '(import <nixpkgs> {}).callPackage ./test.nix {}'
[ { expected = [ "foo:" "foo" null "123" null "123" ]; name = "testUserConfigMatch"; result = [ "foo:" "foo" null "1234" null "1234" ]; } ]
```

Reply:
We could test a few combinations like this even.